### DISCUSSION TITLE
Side panel refresh failure

### DISCUSSION THREAD

**Discussion by Bitello** [Budibase/budibase#9651] [Help] [answered]
Hello!

I customized a side panel so that a certain user profile could, with limitations, edit and add records in a table related to the one I'm working on at the moment.

As a limitation, in the side panel, the "main table" relationship field cannot be edited. This field I pass through a "state".

It happens that the refresh in the "state" change only occurs when the side panel is closed.

![bug01](https://user-images.githubusercontent.com/22456825/218120666-016eab43-c86b-425f-b0eb-543c86197667.gif)

How do I refresh the side panel without having to close it (this is necessary because not changing it leads to registration errors).

Generating an action right after clicking on the link that opens the side panel is not a good option because, when the first click is made, the panel opens correctly, but from then on it locks the panel and needs two clicks to open it again (and with the correct field charge).

How to proceed?

Grateful

**Comment by melohagan** [comment, marked as answer]
Hi @Bitello 

The nesting of components will be important for this one.

Consider the following example where you have a table of *Employees* and a side panel that contains a 'create' form for a new *Job*.

![inline_edit](https://user-images.githubusercontent.com/101575380/219011176-9ae29e00-bb69-40de-bd53-87d5aeea181e.gif)

The way to achieve this, you will have a nested structure along the lines of this:
![Screenshot 2023-02-15 at 11 19 17](https://user-images.githubusercontent.com/101575380/219013379-17ef9a56-5cef-4345-a957-dea3c268b23d.png)

So here the form for the **Jobs** table sits a level above the **Employees** table. Therefore in our table icon component we can define actions as follows:
1. Open the 'New job side panel'
2. **Update Field Value** of the *Assigned* (employee relationship) to the **_id** of the current employee row.

![Screenshot 2023-02-15 at 11 21 03](https://user-images.githubusercontent.com/101575380/219013771-0111239a-ab0c-4331-b80d-bdc6077da687.png)

**Reply by Bitello** [reply]
Hi @melohagan

I don't know if it was some "side effect" of these last updates, but the behavior that was expected when selecting different links started to occur without performing any intervention in the code.

Anyway, your explanation was very useful because, in other scenarios, I was "abusing" the use of "states" when I could be using "Update field value" in using a nested structure properly.

Thanks!

**Reply by melohagan** [reply]
Glad to help! 😄
